packages/core/src/utils/traceData.ts

```diff
@@ -0,0 +1,87 @@
+import type { Client, Scope, Span } from '@sentry/types';
+import {
+  TRACEPARENT_REGEXP,
+  dynamicSamplingContextToSentryBaggageHeader,
+  generateSentryTraceHeader,
+  logger,
+} from '@sentry/utils';
+import { getClient, getCurrentScope } from '../currentScopes';
+import { getDynamicSamplingContextFromClient, getDynamicSamplingContextFromSpan } from '../tracing';
+import { getActiveSpan, getRootSpan, spanToTraceHeader } from './spanUtils';
+
+/**
+ * Extracts trace propagation data from the current span or from the client's scope (via transaction or propagation
+ * context) and serializes it to `sentry-trace` and `baggage` values to strings. These values can be used to propagate
+ * a trace via our tracing Http headers or Html `<meta>` tags.
+ *
+ * This function also applies some validation to the generated sentry-trace and baggage values to ensure that
+ * only valid strings are returned.
+ *
+ * @param span a span to take the trace data from. By default, the currently active span is used.
+ * @param scope the scope to take trace data from By default, the active current scope is used.
+ * @param client the SDK's client to take trace data from. By default, the current client is used.
+ *
+ * @returns an object with the tracing data values. The object keys are the name of the tracing key to be used as header
+ * or meta tag name.
+ */
+export function getTraceData(
+  span?: Span | undefined,
+  scope?: Scope,
+  client?: Client,
+): { 'sentry-trace': string; baggage?: string } {
+  const clientToUse = client || getClient();
+  const scopeToUser = scope || getCurrentScope();
+  const spanToUse = span || getActiveSpan();
+
+  const { dsc, sampled, traceId } = scopeToUser.getPropagationContext();
+  const rootSpan = spanToUse && getRootSpan(spanToUse);
+
+  const sentryTrace = spanToUse ? spanToTraceHeader(spanToUse) : generateSentryTraceHeader(traceId, undefined, sampled);
+
+  const dynamicSamplingContext = rootSpan
+    ? getDynamicSamplingContextFromSpan(rootSpan)
+    : dsc
+      ? dsc
+      : clientToUse
+        ? getDynamicSamplingContextFromClient(traceId, clientToUse)
+        : undefined;
+
+  const baggage = dynamicSamplingContextToSentryBaggageHeader(dynamicSamplingContext);
+
+  const isValidSentryTraceHeader = TRACEPARENT_REGEXP.test(sentryTrace);
+  if (!isValidSentryTraceHeader) {
+    logger.warn('Invalid sentry-trace data. Returning empty "sentry-trace" value');
+  }
+
+  const validBaggage = isValidBaggageString(baggage);
+  if (!validBaggage) {
+    logger.warn('Invalid baggage data. Not returning "baggage" value');
+  }
+
+  return {
+    'sentry-trace': isValidSentryTraceHeader ? sentryTrace : '',
```

Comment:
```suggestion
    'sentry-trace': isValidSentryTraceHeader ? sentryTrace : '',
```
q: why do we want to expose a potentially empty `sentry-trace` but not an empty `baggage`?

Reply:
Let me rewrite this a bit. The general idea is: SDKs picking up tracing data can handle a missing baggage header or tag but not a missing sentry-trace header. So this should always be returned. To be honest, the `sentry-trace` validation logic here is probably not necessary. But I'll rewrite it a bit to return nothing instead of an empty string if `sentry-trace` is invalid.